EIPS/eip-3102.md

```diff
@@ -0,0 +1,189 @@
+---
+eip: 3102
+title: Binary trie structure
+author: Guillaume Ballet (@gballet), Vitalik Buterin (@vbuterin)
+status: Draft
+discussions-to: https://ethresear.ch/t/binary-trie-format/7621
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-09-01
+---
+
+## Simple Summary
+
+A proposal for changing the format of the data-storage structure from hexary to binary, merging the account and storage tries, and using the faster BLAKE2. The goal is to make stateless proofs more wieldy.
+
+## Abstract
+
+This proposal presents a binary structure and merkelization rule for the account and storage tries, which are merged into a single “state” trie. RLP and most of the MPT’s optimizations are dropped to simplify the design. Keccak256 is replaced with BLAKE2-256.
+
+## Motivation
+
+The current design of the Merkle Patricia Trie (MPT) uses an hexary trie. Hexary Merkle trees are more shallow than their binary counterparts, which means less hashing.
+Over the course of the 5 years of Ethereum’s existence, it has become apparent that disk accesses are a greater bottleneck than hashing. Clients are therefore moving away from a storage model in which all internal nodes are stored, in favor of a flat (key, value) storage model first used by turbo-geth, in which the intermediate nodes are recalculated only when needed.
+
+There is a push for making Ethereum easier to use in a stateless fashion. Binary tries make for smaller (~4x) proofs than hexary tries, making it the design of choice for a stateless-friendly Ethereum.
+
+For that same reason, the account and storage tries are merged in order to have a single proof for all changes.
+
+The MPT design is also rife with uncanny optimizations for size, that have a limited effect - at the cost of prohibitive complexity. For example, nesting for children whose RLP is less than 32 bytes saves an estimated 1MB of disk space. A paltry compared to the 300GB required by a fast sync at the time of this writing. These optimizations are a significant source of errors, and therefore a consensus-breaking risk.
+The reliance on RLP has also been criticized for its complexity, while the overhead of a general-purpose encoding scheme isn’t warranted for the rigid structure of a Merkle trie.
+
+The desire to switch the storage model from an hexary trie to a binary trie provides an opportunity to adopt a simpler trie architecture that pushes optimizations from the protocol level down to that of the client implementors.
+
+## Specification
+
+### Conventions
+
+As defined in the yellow paper, $\mathbb{B}$ is the set of all possible byte sequences, and $\mathbb{B}_{32}$ is the set of all byte sequences of length 32. Accordingly, $\mathbb{B}_{l}$ is the set of all byte sequences of length $l$. 
```

Comment:
I don't believe any of the EIP rendering systems support latex.  I recommend using programmer style rather than mathematician style as that will make this specification readable in just about any rendering system (including plain text).